### PR TITLE
Major refactoring of the GeoTables.jl package

### PR DESCRIPTION
**Pull request by eliascarv**
This PR moves the `Data` interface and `MeshData` definitions and tests from Meshes.jl to GeoTables.jl and renames them to `AbstractGeotable` and `GeoTable`.
This PR also deprecates the functions: `load`, `save` and `gadm`, which were moved to the GeoIO.jl package, as well as removes their codes and tests.

### CODE REVIEW

File: src/GeoTables.jl [hunk 1]
```diff
@@ -5,101 +5,21 @@
...
 using Meshes
+using Tables
+using Random
 
-import Tables
-import Meshes
+import Base: values
+import Meshes: partition, partsubsets, unview
+import DataAPI: nrow, ncol
+import StatsBase: sample
```
Comment (juliohm) [inline src/GeoTables.jl:14]: Maybe we should move partition and sample to GeoStatsBase.jl and leave the minimum number of methods in this package. Just the ones that are strictly necessary to manipulate the type as a table.

**Comment by KeithWM** [comment]
I'm a bit surprised this was not given a major version increment. This is a massively breaking change. I now had to search through all PRs to find a mention of "load" to discover why following, .e.g., https://discourse.julialang.org/t/accessing-x-y-values-in-a-geopackage/103196/15 was failing at the very first hurdle. Poor discipline w.r.t. semantic versioning is a real pain to anyone depending on your package, and you should really be more careful.

**Comment by juliohm** [comment]
Definitely an issue of the past. The ecosystem changed so much with so many
breaking changes already that the post from Discourse should be updated
instead.

The load lives in GeoIO.jl for a long time...

Em ter., 27 de fev. de 2024 19:26, KeithWM ***@***.***>
escreveu:

**Comment by KeithWM** [comment]
Indeed, can't change the past, but I was a bit annoyed by this. If the API is constantly changing, it should really still be in 0.x, where a step to 0.x+1 implies it is breaking. Let's hope it's stable now.

**Comment by greimel** [comment]
I'd like to ask you for a bit more care in the future. This a PR was breaking. Code from the README stopped working. (`GeoTables.gadm`). Why didn't you make this version `2.0`?

Even if you made this `2.0` it would have been nice to provide guidance on how to migrate code in the README.

**Comment by juliohm** [comment]
We added this migration information a long time ago in the README as far as
I remember. After months of the release we decided to remove from the
README.

If we knew gadm or GeoTables.jl was heavily used, we would be more careful.
The package has very few stars so we assumed that few people were actually
using it.

Em qua., 6 de mar. de 2024 07:14, Fabian Greimel ***@***.***>
escreveu:

**Comment by greimel** [comment]
As far as I can see the README was last changed in this PR with no migration information. 

Could you point me to the package where `GeoTables.gadm` was moved?

**Comment by juliohm** [comment]
GeoArtifacts.jl is the new home.

Em qua., 6 de mar. de 2024 09:18, Fabian Greimel ***@***.***>
escreveu:

**Pull request merged**